dashboard/app/models/user.rb

```diff
@@ -605,10 +605,43 @@ def user_level_locked?(script_level, level)
     user_level.nil? || user_level.locked?(script_level.stage)
   end
 
+  # Returns the level in this script that the user most recently made progress on
+  def last_progressed_level(script)
+    user_levels.
+      where(script_id: script.id).
+      sort_by(&:updated_at).
+      last
+  end
+
+  # Returns the next script_level for the next progression level in the given
+  # script that hasn't yet been passed, starting its search at the last level we submitted
   def next_unpassed_progression_level(script)
     user_levels_by_level = user_levels_by_level(script)
 
-    script.script_levels.detect do |script_level|
+    user_level = last_progressed_level(script)
```

Comment:
One option would be to do something like this:
```
user_level = user_levels_by_level.values.flatten.sort_by!(&:updated_at).last
```
which avoids another db query, at the expensive of some complexity. The number of user_levels here should be on the order of ~100 at most, so I don't think we need to worry too much about the cost of sorting.

Thoughts?